public/systems_monitoring_graphs.js

```diff
@@ -0,0 +1,1087 @@
+                .attr("class", "diskArea")  // I add the class memoryArea to be able to modify it later on.
+                .attr("fill", "#0E86D4")
+                .attr("fill-opacity", .6)
+                .attr("stroke", "#0E86D4")
+                .attr("stroke-width", 0.2)
+                .attr("d", area);
+
+            // Add brushing
+            let brush = d3.brushX()                 // Add the brush feature using the d3.brush function
+                .extent( [ [0,0], [Width, Height] ] ) // initialise the brush area: start at 0,0 and finishes at width,height: it means I select the whole graph area
+                .on("end", updateChart) // Each time the brush selection changes, trigger the 'updateChart' function
+
+            areaChart
+                .append("g")
+                .attr("class", "brush")
+                .call(brush);
+
+            drawFocus(); 
+
+            // Will hold time set before subsequent brushing trigger `updateChart`
+            let idleTimeout
+            function resetIdleTimeout() { idleTimeout = null; }
+
+            // A function that update the chart for given boundaries
+            function updateChart() {
+                let extent = d3.event.selection;
+
+                // If no selection, back to initial coordinate. Otherwise, update X axis domain
+                if(!extent) {
+                    if (!idleTimeout) return idleTimeout = setTimeout(resetIdleTimeout, 350); // This allows to wait a little bit
+                    x.domain(d3.extent(data, d => d.datetime))
+                } else {
+                    x.domain([ x.invert(extent[0]), x.invert(extent[1]) ])
+                    areaChart.select(".brush").call(brush.move, null) // This remove the grey brush area as soon as the selection has been done
+                }
+
+                // Update axis and area position
+                xAxis.transition().duration(1000).call(d3.axisBottom(x).tickFormat(dayTimeFormat));
+
+                // Rotate X axis ticks
+                xAxis.selectAll("text")
+                    .style("text-anchor", "end")
+                    .attr("dx", "-.8em")
+                    .attr("dy", ".15em")
+                    .attr("transform", "rotate(-65)");
+
+                areaChart
+                    .selectAll("path")
+                    .transition().duration(1000)
+                    .attr("d", area)
+
+                d3.selectAll(".diskXGrid").remove();
+                d3.selectAll(".stopDiskGrid").remove();
+                d3.selectAll(".startDiskGrid").remove();
+
+                svg.append("g")			
+                    .attr("class", "diskXGrid")
+                    .attr("transform", "translate(0," + Height + ")")
+                    .call(d3.axisBottom(x)
+                        .tickSize(-Height)
+                        .tickFormat("")
+                    )
+
+                // Add the hitch gridlines
+                hitchLines.append("g")			
```

Comment:
Low priority, but the hitch lines aren't transitioning with the rest of the graph when brush-zooming

Reply:
Updated thanks.